webui/test/integration/utils/mockServer.ts

```diff
@@ -0,0 +1,180 @@
+    } catch (error) {
+      this.sendErrorResponse(res);
+    }
+  }
+
+  private validateMockResponse(mockResponse: any, requestPath: string) {
+    if (typeof mockResponse.status !== 'number') {
+      throw new Error(`Mock for ${requestPath} missing required 'status' field`);
+    }
+  }
+
+  private buildResponseHeaders(mockResponse: any) {
+    const headers = {
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+      'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+      ...mockResponse.headers
+    };
+
+    if (mockResponse.contentType) {
+      headers['Content-Type'] = mockResponse.contentType;
+    } else if (mockResponse.body) {
+      headers['Content-Type'] = 'application/json';
+    }
+
+    return headers;
+  }
+
+  private sendErrorResponse(res: any) {
+    res.writeHead(500, {
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+    res.end(JSON.stringify({ message: 'Internal server error' }));
+  }
+
+  private handleUnmockedApiRequest(res: any) {
+    res.writeHead(404, {
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*'
+    });
+    res.end(JSON.stringify({ message: 'API endpoint not mocked' }));
+  }
+
+  private handleCorsPreflightRequest(res: any) {
+    res.writeHead(200, {
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+      'Access-Control-Allow-Headers': 'Content-Type, Authorization'
+    });
+    res.end();
+  }
+
+  private handleStaticFileRequest(req: any, res: any) {
+    try {
+      if (!req.url) {
+        this.serveIndexHtml(res);
+        return;
+      }
+
+      let filePath = req.url === '/' ? '/index.html' : req.url;
+      if (filePath?.includes('?')) {
+        filePath = filePath.split('?')[0];
+      }
+
+      if (!filePath || filePath === '/') {
+        filePath = '/index.html';
+      }
+      
+      const fullPath = join(process.cwd(), 'dist', filePath);
+      const content = readFileSync(fullPath);
+      
+      const contentType = this.getContentType(filePath);
+      
+      res.writeHead(200, { 'Content-Type': contentType });
+      res.end(content);
+    } catch (error) {
+      this.serveIndexHtml(res);
+    }
+  }
+
+  private serveIndexHtml(res: any) {
+    try {
+      const indexPath = join(process.cwd(), 'dist', 'index.html');
+      const content = readFileSync(indexPath);
+      res.writeHead(200, { 'Content-Type': 'text/html' });
+      res.end(content);
+    } catch (indexError) {
+      res.writeHead(404);
+      res.end('Not found');
+    }
+  }
+
+  private getContentType(filePath: string): string {
+    if (filePath?.endsWith('.js')) return 'application/javascript';
+    if (filePath?.endsWith('.css')) return 'text/css';
+    if (filePath?.endsWith('.json')) return 'application/json';
+    return 'text/html';
+  }
```

Comment:
Suggest to add 'mime-types' to our dev deps (we already indirectly depends on it).
You can remove this function and just use it like:

```js
const ext = filePath.split('.').pop() || '';
const mimeType = mime.lookup(ext) || 'application/octet-stream';
```